### PR TITLE
feat: add bottom sheet to AddCommentForm

### PR DESCRIPTION
**Pull request by conrad-vanl**
### What does this PR do, or why is it needed?

Adds a bottom sheet editor to the AddCommentForm

See video:
https://www.loom.com/share/f8fe3d4b08544be1a266f0e0d3562550

### How do I test this PR?

Unfortunately, because of the way storybook handles both safe area and keyboard-avoiding crap, this component does not work inside of storybook very well. It works, but very inconsistently. The video above was recorded by essentially replacing `App.js` with:

```jsx
// import { LogBox } from 'react-native';

// import StorybookUI, {
//   loadApollosStories,
//   addApollosProviderDecorator,
//   configure,
// } from '@apollosproject/ui-storybook';
// import Providers from './src/Providers';

// import { loadStories } from './storybook/storyLoader';
// import { enableScreens } from 'react-native-screens';

// enableScreens();

// // Storybook issue. Might be solved by updating Core to RN 63
// LogBox.ignoreLogs(['null is not an object (evaluating \'story.id\')', 'Story with id', 'No permission handler detected.']);

// addApollosProviderDecorator(Providers);

// configure(() => {
//   require('./src/Welcome');
//   loadStories();
//   loadApollosStories();
// }, module);

// export default StorybookUI;
import React from 'react';
import { storiesOf } from '@apollosproject/ui-storybook';
import { View, FlatList, Text, ScrollView } from 'react-native';
import { times } from 'lodash';

import Comment from '@apollosproject/ui-kit/src/Comment';
import { ThemeMixin } from '@apollosproject/ui-kit/src/theme';
import BackgroundView from '@apollosproject/ui-kit/src/BackgroundView';
import AddCommentInput from '@apollosproject/ui-kit/src/AddCommentInput';
import { Providers } from '@apollosproject/ui-kit';
import { NavigationContainer } from '@react-navigation/native';

import { enableScreens } from 'react-native-screens';
enableScreens();

const fakeData = [
  { __typename: 'AddCommentInput' },
  ...times(10, (i) => ({
    __typename: 'Comment',
    profile: {
      image: { uri: `https://picsum.photos/seed/${i}/200` },
      nickName: 'Albert Flores',
    },
    subtitle: 'Anderson Campus',
    onPressLike: () => ({}),
    onPressActionMenu: () => ({}),
    commentText:
      'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit. Exercitation veniam consequat sunt nostrud amet.',
  })),
];

export default () => (
  <NavigationContainer>
    <Providers>
      <AddCommentInput
        showInlinePrompt
        onSubmit={() => {}}
        profile={{
          image: { uri: 'https://picsum.photos/200' },
          nickName: 'Jeff Bridges',
        }}
      />
    </Providers>
  </NavigationContainer>
);
```

### CODE REVIEW

**Comment by conrad-vanl** [comment]
** NOT TESTED ON ANDROID **

Charging my android up tonight and will test tomorrow

**Comment by conrad-vanl** [comment]
@vinnyjth should be working on android now. Going to work on tests / snaps next.

**Comment by vinnyjth** [comment]
So, we'll have to port your native changes to `templates`. (We can merge this PR before that happens) Would you like me to tackle that or do you want to take it?

File: packages/apollos-ui-kit/src/Providers.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { nest } from 'recompose';
+import { BottomSheetModalProvider } from '@gorhom/bottom-sheet';
 
 import { ThemeProvider } from './theme';
 import { LayoutProvider } from './LayoutContext';
 
-export default nest(ThemeProvider, LayoutProvider);
+export default nest(LayoutProvider, ThemeProvider, BottomSheetModalProvider);
```
Comment (vinnyjth) [inline packages/apollos-ui-kit/src/Providers.js:7]: nice!

File: packages/apollos-ui-kit/src/AddCommentInput/AddCommentInput.js [hunk 2]
```diff
@@ -133,45 +170,136 @@ const ShareDisclaimer = styled(
...
+  onSubmit,
+  profile,
+}) => {
+  const [currentText, setCurrentText] = useState('');
   const [confirmModalOpen, setConfirmModalOpen] = useState(false);
   const [share, setShare] = useState('PRIVATE');
 
   const onPressSave = async () => {
     onSubmit && (await onSubmit(currentText, share)); // eslint-disable-line no-unused-expressions
     setConfirmModalOpen(false);
```
Comment (vinnyjth) [inline packages/apollos-ui-kit/src/AddCommentInput/AddCommentInput.js:206]: We should probably close/collapse the bottom sheet here as well right? Right now you can still see the bottom sheet after you finish writing and save your journal.

Comment (conrad-vanl) [reply packages/apollos-ui-kit/src/AddCommentInput/AddCommentInput.js:206]: Ah, good point. Easy fix.

**Comment by conrad-vanl** [comment]
@vinnyjth native changes are pretty easy. Can copy them from this PR, or just follow the instructions on both native-gesture-handler and reanimated. Turns out even though we have native-gesture-handler installed, it's not set up properly on Android.

Actually, I'd probably just replicate the native changes from this PR. It's only on the Android side:

- [ ] https://github.com/ApollosProject/apollos-apps/pull/1807/files#diff-44490a2b58cafdda71f611c01f7439f9a22e9a838d99218349dc546644ad14c5
- [ ] https://github.com/ApollosProject/apollos-apps/pull/1807/files#diff-2022df71cb315c879c5ff1a3c0a416ca95519907b91b860c3a06e4a03058e2d6
- [ ] https://github.com/ApollosProject/apollos-apps/pull/1807/files#diff-add33876b1d5bbcaf5051cec65af0bf4410e7652bc237de7e0d476b4d398886b
- [ ] https://github.com/ApollosProject/apollos-apps/pull/1807/files#diff-599bdc2ff2c16379cc0c0cd17629feea1592399f9c8de895b0c9415f0e11f25c
- [ ] https://github.com/ApollosProject/apollos-apps/pull/1807/files#diff-c902c83c26611c9be70cad41a83ab9534769919d9583dbfa4c50134a7f4047dc
- [ ] https://github.com/ApollosProject/apollos-apps/pull/1807/files#diff-f89af84e82965421c39c8eff98e6a0f036c1e01ca7bcadcf9913f957730288ed
- [ ]

**Comment by conrad-vanl** [comment]
Made lots of improvements to this. Should be money on both android and iOS, or at least I hope so 😬

https://www.loom.com/share/27dd076bff9f4785ac927b644ec4c408

**Comment by vinnyjth** [comment]
@conrad-vanl getting this error trying to view a content single on templates
simulator_screenshot_2F6189D3-5FC3-4CFE-998A-CC4F30C856CF

Digging into it now, but let me know if you have any guesses.

**Comment by vinnyjth** [comment]
Oh, looks like I need the native navigation container 👍

**Comment by conrad-vanl** [comment]
Yes @vinnyjth that should be all :) 

Ddin't realize that those stories were hooked up to render the add comment input. Dope!

In storybook on a device with notches, you'll have to close the story book menu, then you'll see the bottom sheet barely poking up. If you grab it slightly it should "jump" into place. Since storybook stories mount with their size shrunk down when the menu is open, it messes with bottom-sheets view calculations.

File: packages/apollos-ui-kit/src/AddCommentInput/AddCommentInput.js [hunk 3]
```diff
@@ -1,224 +1,113 @@
...
-  ({ theme: { sizing, colors } }) => ({
-    themeSize: sizing.baseUnit * 3,
-    buttonIcon: 'chunky-plus',
-    iconButtonProps: {
-      size: sizing.baseUnit * 0.75,
-      fill: colors.white,
-      iconBackground: colors.action.secondary,
+import Touchable from '../Touchable';
+import styled from '../styled';
+import Navigator from './Navigator';
```
Comment (vinnyjth) [inline packages/apollos-ui-kit/src/AddCommentInput/AddCommentInput.js:15]: This seems to be causing issues :/ 

simulator_screenshot_1DB933D2-1609-41DB-9956-66643DD8EE3D

Any easy fixes you can think of or do you want to pair on this?

**Review by vinnyjth — APPROVED** [review]
Sweet! excited to get this shipped

**Pull request merged**